src/client/src/tscParseLineProtocol.c

```diff
@@ -555,67 +557,100 @@ static int32_t fillDbSchema(STableMeta* tableMeta, char* tableName, SSmlSTableSc
   return TSDB_CODE_SUCCESS;
 }
 
+static int32_t getSuperTableMetaFromLocalCache(TAOS* taos, char* tableName, STableMeta** outTableMeta, SSmlLinesInfo* info) {
+  int32_t     code = 0;
+  STableMeta* tableMeta = NULL;
+
+  SSqlObj* pSql = calloc(1, sizeof(SSqlObj));
+  if (pSql == NULL) {
+    tscError("SML:0x%" PRIx64 " failed to allocate memory, reason:%s", info->id, strerror(errno));
+    code = TSDB_CODE_TSC_OUT_OF_MEMORY;
+    return code;
+  }
+  pSql->pTscObj = taos;
+  pSql->signature = pSql;
+  pSql->fp = NULL;
+
+  registerSqlObj(pSql);
+  char tableNameBuf[TSDB_TABLE_NAME_LEN + TS_ESCAPE_CHAR_SIZE] = {0};
+  memcpy(tableNameBuf, tableName, strlen(tableName));
+  SStrToken tableToken = {.z = tableNameBuf, .n = (uint32_t)strlen(tableName), .type = TK_ID};
+  tGetToken(tableNameBuf, &tableToken.type);
+  bool dbIncluded = false;
+  // Check if the table name available or not
+  if (tscValidateName(&tableToken, true, &dbIncluded) != TSDB_CODE_SUCCESS) {
+    code = TSDB_CODE_TSC_INVALID_TABLE_ID_LENGTH;
+    sprintf(pSql->cmd.payload, "table name is invalid");
+    taosReleaseRef(tscObjRef, pSql->self);
+    return code;
+  }
+
+  SName sname = {0};
+  if ((code = tscSetTableFullName(&sname, &tableToken, pSql, dbIncluded)) != TSDB_CODE_SUCCESS) {
+    taosReleaseRef(tscObjRef, pSql->self);
+    return code;
+  }
+
+  char fullTableName[TSDB_TABLE_FNAME_LEN] = {0};
+  memset(fullTableName, 0, tListLen(fullTableName));
+  tNameExtractFullName(&sname, fullTableName);
+
+  size_t size = 0;
+  taosHashGetCloneExt(UTIL_GET_TABLEMETA(pSql), fullTableName, strlen(fullTableName), NULL, (void**)&tableMeta, &size);
+  taosReleaseRef(tscObjRef, pSql->self);
+
+  STableMeta* stableMeta = tableMeta;
+  if (tableMeta != NULL && tableMeta->tableType == TSDB_CHILD_TABLE) {
+      taosHashGetCloneExt(UTIL_GET_TABLEMETA(pSql), tableMeta->sTableName, strlen(tableMeta->sTableName), NULL,
+                          (void**)stableMeta, &size);
+  }
+
+  if (stableMeta != tableMeta) {
```

Comment:
if retrieved table schema is child table schema, why retry with tableMeta->sTableName?